pkgs/development/python-modules/cmsdials/default.nix

```
@@ -0,0 +1,56 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  poetry-core,
+  pydantic_1,
+  pythonRelaxDepsHook,
+  requests,
+  typing-extensions,
+  pandas,
+  tqdm,
+}:
+
+buildPythonPackage rec {
+  pname = "cmsdials";
+  version = "1.2.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "cms-DQM";
+    repo = "dials-py";
+    rev = "v${version}";
+    hash = "sha256-664FvPdZFU5eTcMadnHfs0ZeiKqTH+Gy2bnhIFQxU2o=";
+  };
+
+  nativeBuildInputs = [
+    poetry-core
+    pythonRelaxDepsHook
+  ];
+
+  propagatedBuildInputs = [
+    pydantic_1
```

```suggestion
    pydantic
```
Please don't propagate the versioned library. It causes a package conflict.

> Only unversioned attributes (e.g. pydantic, but not pypdantic_1) can be included in dependencies, since due to PYTHONPATH limitations we can only ever support a single version for libraries without running into duplicate module name conflicts.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines-contributing-guidelines

---

Should I loosen the major-version restriction of `pydantic` with `relaxPythonDeps`? How could we hint to users about the preferred version?